tests/performance/run_server_performance_test.py

```diff
@@ -0,0 +1,346 @@
+import argparse
+import csv
+import datetime
+import math
+import os
+import random
+import signal
+import subprocess
+import threading
+import time
+from collections import defaultdict
+
+
+VERBOSE = False
+FINISH = False
+PROCESSES = []
+
+
+def finish_test(signum, frame):
+    print('-----> Performance test stops. '
+          'Please wait for stopping all subprocesses. <-----')
+
+    global FINISH
+    FINISH = True
+
+    global PROCESSES
+    for proc in PROCESSES:
+        try:
+            proc.terminate()
+        except OSError:
+            pass
+
+
+signal.signal(signal.SIGINT, finish_test)
+
+
+def return_duration(func):
+    """
+    This decorator makes the applied function return its duration. The original
+    return value gets lost.
+    """
+    def func_wrapper(*args, **kwargs):
+        before = datetime.datetime.now()
+        func(*args, **kwargs)
+        after = datetime.datetime.now()
+        return (after - before).total_seconds()
+
+    return func_wrapper
+
+
+def print_process_output(message, stdout, stderr):
+    global VERBOSE
+
+    if not VERBOSE:
+        return
+
+    print(message)
+    print('-' * 20 + 'stdout' + '-' * 20)
+    print(stdout)
+    print('-' * 20 + 'stderr' + '-' * 20)
+    print(stderr)
+    print('-' * (40 + len('stdout')))
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
```

Comment:
It would be useful to include default values of the argument in the help. Use `argparse.ArgumentDefaultsHelpFormatter` for this:
```py
formatter_class=argparse.ArgumentDefaultsHelpFormatter,
```